prow/cmd/deck/main.go

```diff
@@ -1222,16 +1244,38 @@ func handleProwJob(prowJobClient prowv1.ProwJobInterface) http.HandlerFunc {
 }
 
 // canTriggerJob determines whether the given user can trigger any job.
-func canTriggerJob(user string, cfg *config.RerunAuthConfig) bool {
+func canTriggerJob(user string, pj prowapi.ProwJob, cfg *config.RerunAuthConfig, cli prowgithub.Client) (bool, error) {
 	if cfg.AllowAnyone {
-		return true
+		return true, nil
 	}
 	for _, u := range cfg.AuthorizedUsers {
 		if prowgithub.NormLogin(u) == prowgithub.NormLogin(user) {
-			return true
+			return true, nil
+		}
+	}
+	jobPermissions := pj.Spec.RerunPermissions
+	if jobPermissions.AllowAnyone {
+		return true, nil
+	}
+	for _, u := range jobPermissions.GitHubUsers {
+		if prowgithub.NormLogin(u) == prowgithub.NormLogin(user) {
+			return true, nil
+		}
+	}
+	// if there is no client, no token was provided, so we cannot access the teams
+	if cli == nil {
+		return false, nil
+	}
+	for _, ght := range jobPermissions.GitHubTeams {
+		member, err := cli.TeamHasMember(ght, user)
```

Comment:
This is the only method you use. Consider defining an interface to make this easier to unit test.

Reply:
Done, added a test as well